ecl2df/compdat.py

```diff
@@ -241,6 +255,76 @@ def unrolldf(dframe, start_column="K1", end_column="K2"):
     return unrolled
 
 
+def applywelopen(compdat_df, welopen_df):
+    """Apply WELLOPEN actions to the COMPDAT dataframe.
+
+    Example: COMPDAT and WELOPEN keyword::
+
+      COMPDAT
+        'OP1' 33 44 10 11 'OPEN' /
+      /
+      WELOPEN
+        'OP1' SHUT /
+      /
+
+    will be equal to defining the connections SHUT within COMPDAT.
+
+    Args:
+        compdat_df (pd.DataFrame): Dataframe with unrolled COMPDAT data
+        welopen_df (pd.DataFrame): Dataframe with WELOPEN actions
+
+    Returns:
+        pd.Dataframe, compdat_df now including WELOPEN actions
+
+    """
+    welopen_df = welopen_df.astype(object).where(pd.notnull(welopen_df), None)
+    for _, row in welopen_df.iterrows():
+        if row["I"] and row["J"] and row["K"]:
+            previous_state = compdat_df[
+                (compdat_df["WELL"] == row["WELL"])
+                & (compdat_df["DATE"] <= row["DATE"])
+                & (compdat_df["I"] == row["I"])
+                & (compdat_df["J"] == row["J"])
+                & (compdat_df["K1"] == row["K"])
+                & (compdat_df["K2"] == row["K"])
+            ].drop_duplicates(subset=["I", "J", "K1", "K2"], keep="last")
+        elif row["C1"] or row["C2"]:
+            raise ValueError(
+                "Lumped connections are not supported in the a WELOPEN keyword."
```

Comment:
Typo; "the a"

Reply:
Since this is a ValueError, it will err hard. Which will make it impossible to extract only COMPDAT from a random deck where C1 and C2 is used. Downgrade to WARNING and skip the current KeyWord (or catch the ValueError where `applywelopen()` is performed). (and if exception is to be used, I wonder if there is a NotImplementedError that can be thrown instead?)

Reply:
fixed, downgraded to warning using the logger